### PR TITLE
add fetch instructions to README

### PR DESCRIPTION
**Pull request by llimllib**
It took me a little while to figure out, even though it's simple, so I thought it would be a good addition to the docs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,6 +139,19 @@ See : https://sql-js.github.io/sql.js/examples/GUI/gui.js
 
 #### Loading a database from a server
 
+##### using fetch
+
+```javascript
+  const SQL = await initSqlJs({
+    locateFile: file => `https://path/to/your/dist/folder/dist/${file}`
+  });
+  const dbres = await fetch("/path/to/databse.sqlite");
```
Comment (lovasoa) [inline README.md:148]: You don't need to wait for sql.js to have loaded to start making the request to your database. I would use `Promise.all`, or put the `await`s after the two promise creations.

**Review by lovasoa — CHANGES_REQUESTED** [review]

**Comment by llimllib** [comment]
Something like this is what you'd prefer?

```
  const sqlPromise = initSqlJs({
    locateFile: file => `https://path/to/your/dist/folder/dist/${file}`
  });
  const dataPromise = fetch("/path/to/databse.sqlite").then(res => res.arrayBuffer());
  const [SQL, buf] = await Promise.all([sqlPromise, dataPromise]);
  const db = new SQL.Database(new Uint8Array(buf));
```

I hate those variable names, but dropping the `Promise` bit makes them super unclear to me. Suggestions, both on entirely what I'm doing and on preferable names, welcome

**Comment by llimllib** [comment]
tbh I kind of prefer the non-promised version because the user is probably constructing the SQL object somewhere besides right where they're fetching the data? It's only there for reference really, and a user can make it more efficient if they want?

**Comment by llimllib** [comment]
But of course it's your repo! So whatever makes sense to you wfm

**Comment by lovasoa** [comment]
> ```js
>   const sqlPromise = initSqlJs({
>     locateFile: file => `https://path/to/your/dist/folder/dist/${file}`
>   });
>   const dataPromise = fetch("/path/to/databse.sqlite").then(res => res.arrayBuffer());
>   const [SQL, buf] = await Promise.all([sqlPromise, dataPromise]);
>   const db = new SQL.Database(new Uint8Array(buf));
> ```

This looks good to me. Just drop the extra indentation.

> the user is probably constructing the SQL object somewhere besides right where they're fetching the data

Yes, they will probably have the creation of `sqlPromise` somewhere else. But the await for that promise should probably still be near the code that will use it, not near the code that instantiated it. I don't see a good reason to wait the `.wasm` and the `.sqlite` sequentially rather than in parallel, even in the context of a larger application.

**Comment by llimllib** [comment]
works for me, will update it

**Review by lovasoa — APPROVED** [review]

**Comment by llimllib** [comment]
Thanks!

**Pull request merged**